src/io/syncfile.c

```diff
@@ -492,6 +492,18 @@ MVMObject * MVM_file_open_fh(MVMThreadContext *tc, MVMString *filename, MVMStrin
         MVM_exception_throw_adhoc_free(tc, waste, "Failed to open file %s: %s", fname, uv_strerror(req.result));
     }
 
+    /* Check that we didn't open a directory by accident. If stat fails, just move on. */
```

Comment:
We don't really have a way to check if a failure to call `fstat()` indicates an inability to read from/write to the file. `ENOMEM` (not enough kernel memoy) and `EOVERFLOW` aside, as far as I can see, the documented failure modes on linux should already have triggered when opening the file. So if such a failure after a successful call to `open()` happens at all, we're dealing with a truly exceptional situation. As we can't really do anything about it, we just move on and wait for the next IO operation to fail or succeed as it might.